### PR TITLE
feat: Split long error message into 2 parts for user experience

### PR DESCRIPTION
**Pull request by codemaster08240328**
### SUMMARY

In some case, client get too long error message from BE, and this might cause overlaying error alert issue in FE. Bad User Experience.

This PR splits long error message into 2 parts, and returns `message` and `more_message` as an error.
In the frontend, we need updates to show `more_message` field in `show more` button clicking. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
image

Expecting:
All message below JOB ID section will be hidden by `show more` button.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Pinging @etr2460 and @dpgaspar since you might know best if this is the correct approach to generalize handling these errors (breaking them based on a line break) or if there are edge cases (e.g. other types of error messages) that may be impacted by this.

File: superset/db_engine_specs/base.py [hunk 1]
```diff
@@ -901,6 +901,15 @@ def _extract_error_message(cls, ex: Exception) -> str:
         """Extract error message for queries"""
         return utils.error_msg_from_exception(ex)
 
+    @classmethod
+    def _reformat_error_message(cls, message: str) -> Tuple[str, str]:
```
Comment (mistercrunch) [inline superset/db_engine_specs/base.py:905]: I'm not big on a function that returns a tuple, short we have an object like `DatabaseMessage`, maybe it's just a named tuple or similar. That object may have more than two props. Maybe is has (engine, error_code, message_title, short_message, details) (?)

File: superset/db_engine_specs/base.py [hunk 2]
```diff
@@ -901,6 +901,15 @@ def _extract_error_message(cls, ex: Exception) -> str:
...
+        """Reformat error message for user experience"""
+        splitted_message = re.split("\n+", message, 1)
+        if len(splitted_message) > 1:
+            return splitted_message[0], splitted_message[1]
+
+        return splitted_message[0], ""
```
Comment (mistercrunch) [inline superset/db_engine_specs/base.py:911]: I would implement a no-op in `base.py`, something that does very little and maintains the current behavior as the default behavior. `big_query.py` would do the line parsing/splitting based on exactly how things are formatted there.

File: superset/db_engine_specs/base.py [hunk 3]
```diff
@@ -913,19 +920,24 @@ def extract_errors(
             if match:
                 params = {**context, **match.groupdict()}
                 extra["engine_name"] = cls.engine_name
+                message, more_message = cls._reformat_error_message(message % params)
```
[the remark this answers is not in the document]
Comment (mistercrunch) [reply superset/db_engine_specs/base.py:925]: `more_message` is not a great name, and shuffling tuples around is funky. Let's create a proper simple object (probably namedtuple or dataclass) for this with good names.

File: superset/db_engine_specs/bigquery.py [hunk 4]
```diff
@@ -213,6 +218,52 @@ def fetch_data(
...
+        splitted_message = re.split("\n+", message, 1)
+        if len(splitted_message) > 1:
+            return ErrorMessageType(
+                short_message=splitted_message[0], description=splitted_message[1]
+            )
+
+        return ErrorMessageType(short_message=splitted_message[0], description="")
+
+    @classmethod
+    def extract_errors(
```
Comment (mistercrunch) [inline superset/db_engine_specs/bigquery.py:233]: Feels like this should be added to `superset/db_engine_specs/base.py`, and called somewhere (?) It's hard to review this as is without the larger context of how it's handled for other engines, and where it's called

File: superset/db_engine_specs/bigquery.py [hunk 5]
```diff
@@ -78,6 +78,11 @@ class BigQueryParametersSchema(Schema):
     query = fields.Dict(required=False)
 
 
+class ErrorMessageType(TypedDict):
```
Comment (mistercrunch) [inline superset/db_engine_specs/bigquery.py:81]: Thinking about this object, I think there are 3, maybe 4 things:
1. error_category:str ("BigQuery Error", or "Database Error", "Missing Column" or "Unexpected Error")
2. error_message:str ("Invalid expression, the table 'users' does not include a 'gender' column ")
3. error_details[str:optional] (the expandable section)
4. error_code[str:optional] (3332), error from databases or in other context can have a unique code. This is specialized and may not be used most of the times

**Pull request closed without merging**